x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java

```diff
@@ -460,15 +463,31 @@ public void resolveAsSeparateMappings(String indexWildcard, String javaRegex, bo
         FieldCapabilitiesRequest fieldRequest = createFieldCapsRequest(indexWildcard, includeFrozen);
         client.fieldCaps(fieldRequest,
                 ActionListener.wrap(
-                        response -> listener.onResponse(
-                                separateMappings(typeRegistry, indexWildcard, javaRegex, response.getIndices(), response.get())),
+                        response -> {
+                            String[] indicesList = response.getIndices();
+                            GetAliasesRequest aliasRequest = new GetAliasesRequest()
+                                    .local(true)
+                                    .aliases("*")
+                                    .indices(indicesList)
+                                    .indicesOptions(IndicesOptions.lenientExpandOpen());
+                            client.admin().indices().getAliases(aliasRequest, wrap(aliases ->
+                                listener.onResponse(separateMappings(typeRegistry, javaRegex, indicesList, response.get(),
+                                        aliases.getAliases())),
+                                ex -> {
+                                    if (ex instanceof IndexNotFoundException || ex instanceof ElasticsearchSecurityException) {
+                                        listener.onResponse(separateMappings(typeRegistry, javaRegex, indicesList, response.get(), null));
+                                    } else {
+                                        listener.onFailure(ex);
+                                    }
+                                }));
+                        },
                         listener::onFailure));
 
     }
     
-    public static List<EsIndex> separateMappings(DataTypeRegistry typeRegistry, String indexPattern, String javaRegex, String[] indexNames,
-            Map<String, Map<String, FieldCapabilities>> fieldCaps) {
-        return buildIndices(typeRegistry, indexNames, javaRegex, fieldCaps, Function.identity(), (s, cap) -> null);
+    public static List<EsIndex> separateMappings(DataTypeRegistry typeRegistry, String javaRegex, String[] indexNames,
+            Map<String, Map<String, FieldCapabilities>> fieldCaps, ImmutableOpenMap<String, List<AliasMetaData>> aliases) {
```

Comment:
What was wrong with `Map` ?

Reply:
ImmutableOpenMap is not actually a `java.util.Map`.